elastic/kibana
# [Fleet] Use API key for standalone agent onboarding

## Summary

Closes https://github.com/elastic/kibana/issues/167218

This PR replaces the username and password with API key config in standalone agent onboarding.

Note: the Observability Logs onboarding (at `app/observabilityOnboarding/systemLogs/?category=logs`) shows the encoded API key, I thought it would make more sense to show it in the Beats format (see https://www.elastic.co/guide/en/fleet/current/grant-access-to-elasticsearch.html).

### Testing

Below are the steps for testing standalone agent install from the main Fleet UI (Agents table). I am not entirely sure how to test [the component used in CreatePackagePolicyPage](https://github.com/elastic/kibana/blob/main/x-pack/plugins/fleet/public/applications/fleet/sections/agent_policy/create_package_policy_page/multi_page_layout/components/page_steps/install_agent/install_agent_standalone.tsx), so I'd appreciate a comment on that. 🙏 

⚠️ Ideally, this should also be tested in serverless config.

1. Open the Add agent flyout and select Standalone. Notice that the agent yaml config contains an `api_key` entry instead of `username` and `password`.
4. Click the new "Create API key" button. This should create a new API key and that can be copied (you can find this key under Stack Management -> Api keys, it should be called `standalone_agent-{randomString}`).
5. Check that the `{API_KEY}` placeholder in the agent yaml was updated with the API key value.
6. Download an Elastic Agent, e.g. on a VM.
7. Modify the `elastic-agent.yml` file of the agent to the yaml from the UI. If using a Multipass VM, you can for instance download it from the UI and copy it using `multipass transfer <localFileLocation> <VMName>:./<elasticAgentDirectory>`.
8. Install the agent as standalone: `sudo ./elastic-agent install` (answer `n` when asked about enrolling in Fleet).
9. Check the agent status and logs with `sudo elastic-agent status` and `sudo elastic-agent logs`.
10. In the UI, go to Discover and search for the agent host name in the logs, it should appear.

### Screenshots

Current behaviour (on `main`):
<img width="799" alt="Screenshot 2024-06-28 at 10 38 44" src="https://github.com/elastic/kibana/assets/23701614/059bf45c-66db-447b-bf01-4226c64ce2a6">

With this change:
<img width="799" alt="Screenshot 2024-06-28 at 10 31 35" src="https://github.com/elastic/kibana/assets/23701614/5853fbf3-221e-48fd-9d13-aececa78f389">

After having clicked "Create API key":
<img width="799" alt="Screenshot 2024-06-28 at 10 31 58" src="https://github.com/elastic/kibana/assets/23701614/df066040-cdbe-4fd5-a508-7093af86e85c">

### Checklist
